### PR TITLE
Add Mapbox GL renderer

### PR DESCRIPTION
**Pull request by bartvde**
Now ready for review

this decrease in coverage seems to be the maximum possible, given that we cannot test Mapbox GL JS under node.

### CODE REVIEW

File: src/components/mapboxgl.js [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+        const layer_name = features[i].layer.id;
+        if (!features_by_layer[layer_name]) {
+          features_by_layer[layer_name] = [];
+        }
+        features_by_layer[layer_name] = features[i];
+      }
+      resolve([features_by_layer]);
+    });
+    // no xy and hms properties here
+    const coordinate = {
```
Comment (theduckylittle) [inline src/components/mapboxgl.js]: is this better than just using `const coordinate = [lat, lon];` ?

**Comment by theduckylittle** [comment]
this all works and I approve, recognizing the drop in coverage. My only concern is not fixable here which is that I wish we were able to  _choose_ which libraries are included based on the renderer. As is, we'll be pulling a lot of code down which may never be used.

File: src/components/mapboxgl.js (hunk 1 above)
Comment (bartvde) [reply src/components/mapboxgl.js]: I was attempting to keep the same structure as here: https://github.com/boundlessgeo/sdk/blob/master/src/components/map.js#L1248:L1252

**Comment by bartvde** [comment]
Thanks @duck, but given that we have marked them as peerDependencies and all peerDependencies are optional, it shoudn't mean everything will get pulled in automatically? I.e. if someone writes an app that uses the openlayers map, they include ```src/components/map``` and they make sure they have the ```ol``` peerDependency? If they want to use the mapbox renderer, they include ```src/components/mapboxgl``` and include the mapbox-gl-js, draw and turf peerDeps?

File: src/components/mapboxgl.js (hunk 1 above)
Comment (theduckylittle) [reply src/components/mapboxgl.js]: Oh sure, that makes total sense then.

**Pull request merged**